pkg/client/cache/delta_fifo.go

```diff
@@ -100,6 +100,11 @@ type DeltaFIFO struct {
 	items map[string]Deltas
 	queue []string
 
+	// firstCalled is used to record the first method called by caller
+	firstCalled int
+	// count is the number of items inserted by the first call of Replace()
+	count int
```

Comment:
May I suggest, instead:

```go
populated bool
initialPopulationCount int
```

Then all of add/update/delete set populated to true; and replace does

```go
if !f.populated {
  f.populated = true
  f.initialPopulationCount = len(list)
}
```

HasSynced does

```go
return f.populated && f.initialPopulationCount == 0
```

I think this is simpler than keeping track of the exact method first called, I'd rather not do that unless needed.